.github/workflows/delete-pool-on-pr-close.yaml

```diff
@@ -0,0 +1,63 @@
+name: Tear down Batch pool
+
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+
+  delete-pool:
+    runs_on: cdc-cdcgov
+    name: Delete Batch pool
+
+  steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - name: Get commit message
+      id: commit-message
+      run: echo "message=$(git log -1 --pretty=%s HEAD)" >> $GITHUB_OUTPUT
+
+      # From: https://stackoverflow.com/a/58035262/2097171
+    - name: Extract branch name
+      shell: bash
+      run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
+      id: branch-name
+
+    - name: Figure out tag (either latest if it is main or the branch name)
+      id: image-tag
+      run: |
+        if [ "${{ steps.branch-name.outputs.branch }}" = "main" ]; then
+          echo "tag=latest" >> $GITHUB_OUTPUT
+        else
+          echo "tag=${{ steps.branch-name.outputs.branch }}" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Install az CLI
```

Comment:
Is this not installed on the cfa-cdcgov runner already?

Reply:
I didn't think so? We install it in the other actions in this repo. Would be convenient if it was!

Reply:
Yeah I'm not sure -- @jkislin do you know?

Reply:
Should be. But also, just do one of these for much better reproducibility/environment independence:

1. Use a pre-baked official action with exactly the same functionality: https://github.com/marketplace/actions/azure-cli-action
2. Use a container that already has it installed:   
https://docs.github.com/en/actions/writing-workflows/choosing-where-your-workflow-runs/running-jobs-in-a-container

You can see:
```yaml
# File: .github/workflows/workflow.yml

on: [push]

name: AzureCLISample

jobs:

  build-and-deploy:
    runs-on: ubuntu-latest
    steps:

    - name: Azure Login
      uses: azure/login@v2
      with:
        creds: ${{ secrets.AZURE_CREDENTIALS }}

    - name: Azure CLI script
      uses: azure/cli@v2
      with:
        azcliversion: latest
        inlineScript: |
          az account show
          az storage -h
```